### PR TITLE
Feature/sandbox

### LINKED ISSUE #129: User Scripts in Auto Responses
Add the ability to write / run scripts in autoresponses.

### PR DESCRIPTION
**Pull request by jammiess**
## Description

Adds a service for running arbitrary user scripts. This will extend the autoresponse system by allowing users to implement their own logic for how they should act through a starlark script.

This PR will create a new docker environment where a gRPC service will reside. This keeps the scripts in an isolated environment and starlark is a limited dialect of python which will both help to ensure security and privacy when running user scripts.

## Checklist

- [X] [Linked PR to corresponding issue in this repo](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue)
- [ ] Manually tested locally
- [X] Additional issue created in [docs repo](https://github.com/architus/docs.archit.us/issues) (if necessary)

### CODE REVIEW

**Comment by jammiess** [comment]
@johnyburd How do I remove the auto generated python files from the lint check?

**Comment by jammiess** [comment]
@jammiess Reminder to make the shorthand versions of the channel, author, and message variables

File: docker-compose.yml [hunk 1]
```diff
@@ -108,5 +108,16 @@ services:
       - webnet
     env_file:
       - architus.env
+  sandbox:
+    image: johnyburd/feature-gate:${TAG:-latest}
```
Comment (johnyburd) [inline docker-compose.yml]: 🤔

File: docker-compose.yml [hunk 2]
```diff
@@ -108,5 +108,16 @@ services:
...
+    build:
+      context: ./
+      dockerfile: ./sandbox/Dockerfile
+    ports:
+      - "1337:1337"
```
Comment (johnyburd) [inline docker-compose.yml]: this shouldn't be exposed to the host

File: docker-compose.yml [hunk 3]
```diff
@@ -108,5 +108,16 @@ services:
...
+    networks:
+      - webnet
+    env_file:
+      - architus.env
```
Comment (johnyburd) [inline docker-compose.yml]: if the service doesn't use any variables no point in risking leaking them

File: lib/response_grammar/response.py [hunk 4]
```diff
@@ -243,6 +248,13 @@ def parse(string):
                 node.parent = curr
                 curr.children.append(node)
                 i += 9
+            elif m := eval_token.search(string[i:]):
+                node = Node()
+                node.type = NodeType.Eval
+                node.text = m.group(1)
+                node.parent = curr
+                curr.children.append(node)
+                i += m.end()
```
Comment (johnyburd) [inline lib/response_grammar/response.py:257]: wow this is very good code :)

File: sandbox/rpc/go.sum [hunk 5]
```diff
@@ -0,0 +1,85 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (johnyburd) [inline sandbox/rpc/go.sum:1]: is this file supposed to be commited? it looks gross

File: shard/src/auto_response.py [hunk 6]
```diff
@@ -182,6 +183,37 @@ async def resolve_resp(self, node, match, msg, content=None, reacts=None):
...
+        elif (node.type == NodeType.Eval):
+            output = await self.bot.sandbox_client.RunStarlarkScript(
+                message.StarlarkScript(
+                    script=node.text,
+                    trigger_message=message.Message(
+                        clean=msg.clean_content,
+                        content=msg.content,
+                        id=msg.id
+                    ),
+                    author=message.Author(
```
Comment (johnyburd) [inline shard/src/auto_response.py:195]: wow more excellent code

**Review by johnyburd — CHANGES_REQUESTED** [review]

**Comment by johnyburd** [comment]
also fix your linting errors

File: lib/response_grammar/response.py (hunk 4 above)
Comment (jammiess) [reply lib/response_grammar/response.py:257]: Do we need to git blame that?

**Comment by jammiess** [comment]
@johnyburd look at the message you were tagged in before asking me to fix lints

File: sandbox/rpc/go.sum (hunk 5 above)
Comment (jammiess) [reply sandbox/rpc/go.sum:1]: I have no clue. But other go repos track it so...

**Comment by johnyburd** [comment]
> @johnyburd look at the message you were tagged in before asking me to fix lints

.flake8

**Review by johnyburd — APPROVED** [review]

**Pull request merged**